cardano-api/src/Cardano/Api/Eras.hs

```diff
@@ -200,6 +199,30 @@ instance Eq AnyCardanoEra where
         Nothing   -> False
         Just Refl -> True -- since no constructors share types
 
+instance Bounded AnyCardanoEra where
+   minBound = AnyCardanoEra ByronEra
+   maxBound = AnyCardanoEra AlonzoEra
+
+instance Enum AnyCardanoEra where
+
+   -- [e..] = [e..maxBound]
+   enumFrom e = enumFromTo e maxBound
+
+   fromEnum = \case
+      AnyCardanoEra ByronEra   -> 0
+      AnyCardanoEra ShelleyEra -> 1
+      AnyCardanoEra AllegraEra -> 2
+      AnyCardanoEra MaryEra    -> 3
+      AnyCardanoEra AlonzoEra  -> 4
+
+   toEnum = \case
+      0 -> AnyCardanoEra ByronEra
+      1 -> AnyCardanoEra ShelleyEra
+      2 -> AnyCardanoEra AllegraEra
+      3 -> AnyCardanoEra MaryEra
+      4 -> AnyCardanoEra AlonzoEra
+      n -> error $ "AnyCardanoEra.toEnum: bad argument: " <> show n
```

Comment:
A better error message would be: `"AnyCardanoEra.toEnum: " <> show n <> " does not correspond to any known enumerated era." `